adapter/internal/grpc-client/grpcExecutor.go

```diff
@@ -0,0 +1,100 @@
+/*
+ *  Copyright (c) 2022, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package client
+
+import (
+	"crypto/tls"
+	"fmt"
+	"time"
+
+	"github.com/wso2/apk/adapter/config"
+	logger "github.com/wso2/apk/adapter/internal/loggers"
+	"github.com/wso2/apk/adapter/pkg/logging"
+	"github.com/wso2/apk/adapter/pkg/tlsutils"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	grpcStatus "google.golang.org/grpc/status"
+) 
+
+
+
+type RetryPolicy struct {
+	// Maximum number of time a failed grpc call will be retried. Set negative value to try indefinitely.
+	MaxAttempts int;
+	// Time delay between retries. (In milli seconds)
+	BackOffInMilliSeconds int;
+}
+
+func GetConnection() (*grpc.ClientConn, error){
+	conf, _ := config.ReadConfigs()
```

Comment:
no need. if there were errors while parsing configs, it would have panicked in an earlier stage. We don't need to return an error from config.ReadConfigs() here at all, I have fixed this using my PR, so let's ignore it here.